app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.kt

```diff
@@ -0,0 +1,739 @@
+                text = roomName
+                setTextColor(titleTextColor)
+                textSize = titleTextSize
+            }
+            roomTitlesRowLayout.addView(roomTitle)
+        }
+    }
+
+    /**
+     * Jump to current time or session, if we are on today's session list
+     */
+    private fun scrollToCurrent(boxHeight: Int) {
+        if (sessionId != null) {
+            return
+        }
+        val currentDayIndex = scheduleData!!.dayIndex
+        if (currentDayIndex != MyApp.dateInfos.indexOfToday) {
+            return
+        }
+        var columnIndex = -1
+        val layoutRootView = requireView()
+        if (!layoutRootView.context.isLandscape()) {
+            val horizontalSnapScrollView = layoutRootView.findViewById<HorizontalSnapScrollView>(R.id.horizScroller)
+            columnIndex = horizontalSnapScrollView.columnIndex
+            MyApp.LogDebug(LOG_TAG, "y pos = $columnIndex")
+        }
+        val nowMoment = Moment.now()
+        val scrollAmount = scrollAmountCalculator!!.calculateScrollAmount(
+            conference = conference!!,
+            dateInfos = MyApp.dateInfos,
+            scheduleData = scheduleData!!,
+            nowMoment = nowMoment,
+            currentDayIndex = currentDayIndex,
+            boxHeight = boxHeight,
+            columnIndex = columnIndex
+        )
+        layoutRootView.requireViewByIdCompat<NestedScrollView>(R.id.verticalScrollView).apply {
+            scrollTo(0, scrollAmount)
+            post { scrollTo(0, scrollAmount) }
+        }
+    }
+
+    private fun setBell(session: Session) {
+        val verticalScrollView = requireView().findViewById<NestedScrollView?>(R.id.verticalScrollView)
+            ?: return
+        val sessionView = verticalScrollView.findViewWithTag<View?>(session)
+            ?: return
+        val bellView = sessionView.findViewById<ImageView?>(R.id.session_bell_view)
+            ?: return
+        bellView.isVisible = session.hasAlarm
+    }
+
+    private fun scrollTo(session: Session) {
+        val height = getNormalizedBoxHeight(displayDensityScale)
+        val pos = scrollAmountCalculator!!.calculateScrollAmount(conference!!, session, height)
+        MyApp.LogDebug(LOG_TAG, "Position is $pos")
+        val layoutRootView = requireView()
+        layoutRootView.requireViewByIdCompat<NestedScrollView>(R.id.verticalScrollView).apply {
+            post { scrollTo(0, pos) }
+        }
+        val horizontalSnapScrollView = layoutRootView.findViewById<HorizontalSnapScrollView?>(R.id.horizScroller)
+        if (horizontalSnapScrollView != null) {
+            val horizontalPos = scheduleData!!.findRoomIndex(session)
+            MyApp.LogDebug(LOG_TAG, "Scroll horizontal to $horizontalPos")
+            horizontalSnapScrollView.post { horizontalSnapScrollView.scrollToColumn(horizontalPos, false) }
+        }
+    }
+
+    private fun chooseDay(chosenDay: Int) {
+        if (chosenDay + 1 != mDay) {
```

Comment:
:thinking: Yoda?

Reply:
See [here](https://en.wikipedia.org/wiki/Yoda_conditions).

Reply:
None of the sides of the conditional statement are constant. Therefore, I won't say the _Yoda_ applies here.

Reply:
Also if you change the order it would be more consistent with other places in the repo or this PR. Look for example in NavigationMenuEntriesGenerator.kt#55 in the changes here...similar case but reversed order

Reply:
I had look here again. The reason why `dayIndex` is 1-indexed is that the schedule XML of day 1 contains the `index='1'` attribute. The corresponding `itemPosition` of the day selection dropdown is 0-indexed. Therefore, it makes sense to me to add one to `choosenDay` (`itemPosition`) to transform it into _schedule-XML-value-space_.

In general it looks like as if one should decouple the value-space of the schedule XML from what the app uses above the database layer. But that is a topic for a separate pull request.

For now, I added a few more comments/KDoc to clarify the offset. See 2feb7521a9d39a94c29cceb8ba55058a06b290c8.

Reply:
Okay that's fine by me